clients/drcachesim/tools/opcode_mix.h

```diff
@@ -134,32 +140,23 @@ class opcode_mix_t : public analysis_tool_t {
         std::unordered_map<uint, int64_t> category_counts_;
     };
 
-    struct worker_data_t {
```

Comment:
Your library has per-shard maps?  It should change to per-worker?  Can make a difference: parallel analysis with 12 workers over 2000 threads: 12 maps instead of 2000.

Reply:
It is certainly extra memory usage to maintain this per-shard instead of per-worker. But I thought that's the right thing to do. Correct me if I'm wrong: for JIT traces, maintaining the cache per-worker means that some shards may get a stale encoding?

Also, the extra memory usage won't be concurrent, as the workers process shards one after the other? https://github.com/DynamoRIO/dynamorio/blob/9931511abf1745e46c2666c1601eb098eb3e0e92/clients/drcachesim/analyzer.cpp#L675

So it shouldn't affect peak memory usage?

Reply:
Hmm, yes, that does seem correct: it has to be per-shard for JIT code in case there is interleaving.  OK, so this worker scheme was not entirely correct.  Please add a note to the PR description since this might confuse future readers.

Memory usage: you mean it will call shard exit on the thread exit record before moving to the next one: but will your library free its map on shard exit?  No: opcode_mix does not free the shards until aggregation at the very end.  Should we put in some mechanism to free earlier w/o throwing out the other shard data it wants to keep?

Reply:
Added a clear_cache API invoked in parallel_shard_exit.